BUG: mcp-add tool causes panic with nil pointer dereference and crashes gateway

## Description
When using the `mcp-add` tool to add the `docker` server dynamically, the MCP Gateway crashes with a panic.

## Environment
- Docker Desktop: 4.55.0 (213807)
- OS: Windows 10 (10.0.26200)
- MCP Client: Cursor IDE
- Transport: stdio

## Steps to Reproduce
1. Start Docker Desktop
2. Configure Cursor with MCP Gateway (docker mcp gateway stdio)
3. Open Cursor Agent
4. Use mcp-find tool - works correctly
5. Use mcp-add with {"name": "docker", "activate": true} - crashes

## Expected Behavior
The docker server should be added to the session.

## Actual Behavior
Gateway crashes with panic and all subsequent calls return "Not connected".

## Error Log
panic: runtime error: invalid memory address or nil pointer dereference
github.com/docker/mcp-gateway/pkg/gateway/mcpadd.go:188

## Workaround
Pre-enable the server manually with docker mcp server enable docker before starting the gateway. But calling mcp-add dynamically still crashes.

--- a/pkg/gateway/mcpadd.go
+++ b/pkg/gateway/mcpadd.go
@@ -151,10 +151,10 @@ func addServerHandler(g *Gateway, clientConfig *clientConfig) mcp.ToolHandler {
 
 		// If secrets or config are missing, handle based on client type
 		if len(missingSecrets) > 0 || len(missingConfig) > 0 {
-			// Check if the client is nanobot
+			// Safely determine client name (InitializeParams may be nil for some transports)
 			clientName := ""
-			if req.Session.InitializeParams().ClientInfo != nil {
-				clientName = req.Session.InitializeParams().ClientInfo.Name
+			if init := req.Session.InitializeParams(); init != nil && init.ClientInfo != nil {
+				clientName = init.ClientInfo.Name
 			}
 
 			if clientName == "nanobot" && len(missingSecrets) > 0 {
@@ -256,17 +256,31 @@ func addServerHandler(g *Gateway, clientConfig *clientConfig) mcp.ToolHandler {
 			}
 		}
 
-		// Register DCR client and make sure OAuth provider is started if this is a remote OAuth server
-		if g.McpOAuthDcrEnabled && serverConfig != nil && serverConfig.Spec.IsRemoteOAuthServer() {
-			authorized, oauthText := g.getRemoteOAuthServerStatus(ctx, serverName, req, shouldSendTools)
-			if !authorized {
-				return &mcp.CallToolResult{
-					Content: []mcp.Content{&mcp.TextContent{
-						Text: oauthText,
-					}},
-				}, nil
+		// Handle OAuth DCR only when the client supports elicitation (e.g. not stdio-based clients)
+		if g.McpOAuthDcrEnabled &&
+			serverConfig != nil &&
+			serverConfig.Spec.IsRemoteOAuthServer() {
+
+			init := req.Session.InitializeParams()
+			if init != nil &&
+				init.Capabilities != nil &&
+				init.Capabilities.Elicitation != nil {
+
+				authorized, oauthText := g.getRemoteOAuthServerStatus(
+					ctx,
+					serverName,
+					req,
+					shouldSendTools,
+				)
+				if !authorized {
+					return &mcp.CallToolResult{
+						Content: []mcp.Content{&mcp.TextContent{
+							Text: oauthText,
+						}},
+					}, nil
+				}
+				responseText = oauthText
 			}
-			responseText = oauthText
 		}
 
 		return &mcp.CallToolResult{
@@ -406,8 +420,11 @@ func (g *Gateway) getRemoteOAuthServerStatus(ctx context.Context, serverName str
 		g.startProvider(ctx, serverName)
 	}
 
-	// Check if current serverSession supports elicitations
-	if req.Session.InitializeParams().Capabilities != nil && req.Session.InitializeParams().Capabilities.Elicitation != nil {
+	// Proceed with elicitation only if the client supports it
+	init := req.Session.InitializeParams()
+	if init != nil &&
+		init.Capabilities != nil &&
+		init.Capabilities.Elicitation != nil {
 		// Elicit a response from the client asking whether to open a browser for authorization
 		elicitResult, err := req.Session.Elicit(ctx, &mcp.ElicitParams{
 			Message: fmt.Sprintf("Would you like to open a browser to authorize the '%s' server?", serverName),

